src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/ModuleScaffold.kt

```diff
@@ -0,0 +1,122 @@
+    private var pitch = 0f
+    private var bothRotations: Rotation? = null
+    private var keepRotation = false
+
+    val repeatable = repeatable {
+        mc.timer.timerSpeed = timer
+
+        val blockPos = BlockPos(player.pos).down()
+
+        // Gipsy KeepRotation
+        if (keepRotation)
+            RotationManager.aimAt(bothRotations!!, configurable = rotations)
+
+        if (!world.getBlockState(blockPos).material.isReplaceable)
+            return@repeatable
+
+        // If there isn't any block detected
+        if (findBlock() == -1)
+            return@repeatable
+
+        // Change slot
+        slot = findBlock()
+        player.inventory.selectedSlot = slot
+
+        // Place block
+        place(blockPos)
+    }
+
+    private fun place(blockPos: BlockPos) {
+        val eyes = player.eyesPos
+
+        for (direction in Direction.values()) {
+            val neighbor = blockPos.offset(direction)
+            val opposite = direction.opposite
+
+            // If block can be clicked
+            if (world.getBlockState(neighbor).getOutlineShape(world, neighbor) == VoxelShapes.empty())
+                continue
+
+            val hitVec = Vec3d.ofCenter(neighbor).add(Vec3d.of(opposite.vector).multiply(0.5))
+
+            if (eyes.squaredDistanceTo(hitVec) > 18)
+                continue
+
+            // Make yaw + pitch based on hitVec and eyes
+            val diffX = hitVec.x - eyes.x
+            val diffY = hitVec.y - eyes.y
+            val diffZ = hitVec.z - eyes.z
+            val diffXZ = sqrt(diffX * diffX + diffZ * diffZ)
+
+            val rotation = Rotation(
+                MathHelper.wrapDegrees(Math.toDegrees(atan2(diffZ, diffX)).toFloat() - 90f),
+                MathHelper.wrapDegrees(-Math.toDegrees(atan2(diffY, diffXZ)).toFloat())
+            )
+
+            // Rotate server-sided
+            keepRotation = true
+            RotationManager.aimAt(rotation, configurable = rotations)
+
+            // Save yaw + pitch to variables to make KeepRotation the gipsy way
+            yaw = rotation.yaw
+            pitch = rotation.pitch
+            bothRotations = rotation
+
+            // Send right click
+            interaction.interactBlock(
+                player,
+                world,
+                Hand.MAIN_HAND,
+                BlockHitResult(hitVec, direction, blockPos, false)
+            )
+            interaction.interactItem(player, world, Hand.MAIN_HAND)
+
+            if (swing)
+                player.swingHand(Hand.MAIN_HAND)
+            else
+                network.sendPacket(HandSwingC2SPacket(Hand.MAIN_HAND))
+        }
+    }
+
+    // Simple hot bar block detection
+    private fun findBlock(): Int {
+        for (i in 0..8) {
+            val itemStack = player.inventory.getStack(i).item
+            if (itemStack is BlockItem) {
+                val blockType = itemStack.block
+                if (blockType !is FallingBlock || blockType !is GrassBlock)
```

Comment:
This expression is always true since there is no block that is a `FallingBlock` and a `GrassBlock` at the same time. You can fix this by replacing the `||` with `&&`